brainiak/searchlight/searchlight.py

```diff
@@ -0,0 +1,325 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#               http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from mpi4py import MPI
+import numpy as np
+
+"""Distributed searchlight
+Given a function and a list of volumes, this applies the function
+to all voxels.
+"""
+
+# Author: Michael Anderson
+
+__all__ = [
+    "Searchlight",
+]
+
+
+class Searchlight:
+    """ Class for searchlight.
+    A searchlight is a computation that is applied in a sliding
+    window to subsets of voxels across a volume. The radius of the
+    searchlight specifies how many voxels are involved in each
+    computation. A user-provided function is applied to all voxels
+    within the radius, provided that the mask was non-zero at that
+    point in the volume. The ouput of the searchlight analysis is
+    another volume which contains the user-provided function's
+    output at each point.
+
+    Parameters
+    ----------
+    None
+
+    Attributes
+    ----------
+    None
+    """
+
+    def __init__(self):
+        pass
+
+    def _get_subarray(self, data, idx, rad):
+        """ Return a subarray with radius 'rad', centered around 'idx' with the
+                same list structure as 'data'.
+
+        Parameters
+        ----------
+        data: A list of 4D numpy.ndarray objects which must have at least one
+        element. The computation is applied to this data.
+
+        idx: Three-element tuple containing the current center of the
+        searchlight
+
+        rad: Odd positive integer indicating the radius of the searchlight
+        cube.
+
```

Comment:
There should be a description for the return value, in case it is not obvious what it represents. Otherwise, the function name itself is an indicator for what the return represents, just like for declarations in C/Java.